specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json

```diff
@@ -0,0 +1,648 @@
+         }
+      },
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/azureFirewalls":{  
+         "get":{  
+            "tags":[  
+               "AzureFirewalls"
+            ],
+            "operationId":"AzureFirewalls_List",
+            "description":"Lists all Azure Firewalls in a resource group.",
+            "parameters":[  
+               {  
+                  "name":"resourceGroupName",
+                  "in":"path",
+                  "required":true,
+                  "type":"string",
+                  "description":"The name of the resource group."
+               },
+               {  
+                  "$ref":"#/parameters/ApiVersionParameter"
+               },
+               {  
+                  "$ref":"#/parameters/SubscriptionIdParameter"
+               }
+            ],
+            "responses":{  
+               "200":{  
+                  "description":"Success. The operation returns a list of AzureFirewall resources.",
+                  "schema":{  
+                     "$ref":"#/definitions/AzureFirewallListResult"
+                  }
+               }
+            },
+            "x-ms-examples":{  
+               "List all Azure Firewalls for a given resource group":{  
+                  "$ref":"./examples/AzureFirewallListByResourceGroup.json"
+               }
+            },
+            "x-ms-pageable":{  
+               "nextLinkName":"nextLink"
+            }
+         }
+      },
+      "/subscriptions/{subscriptionId}/providers/Microsoft.Network/azureFirewalls":{  
+         "get":{  
+            "tags":[  
+               "AzureFirewalls"
+            ],
+            "operationId":"AzureFirewalls_ListAll",
+            "description":"Gets all the Azure Firewalls in a subscription.",
+            "parameters":[  
+               {  
+                  "$ref":"#/parameters/ApiVersionParameter"
+               },
+               {  
+                  "$ref":"#/parameters/SubscriptionIdParameter"
+               }
+            ],
+            "responses":{  
+               "200":{  
+                  "description":"Success. The operation returns a list of AzureFirewall resources.",
+                  "schema":{  
+                     "$ref":"#/definitions/AzureFirewallListResult"
+                  }
+               }
+            },
+            "x-ms-examples":{  
+               "List all Azure Firewalls for a given subscription":{  
+                  "$ref":"./examples/AzureFirewallListBySubscription.json"
+               }
+            },
+            "x-ms-pageable":{  
+               "nextLinkName":"nextLink"
+            }
+         }
+      }
+   },
+   "definitions":{  
+      "AzureFirewallIPConfigurationPropertiesFormat":{  
+         "properties":{  
+			"privateIPAddress": {
+			  "type": "string",
+			  "description": "PrivateIPAddress of the network interface IP Configuration. This field is populated in the output."
```

Comment:
This is actually the front end IP address of the Internal Loadbalancer. we can check with Yair on how to expose this and if we should say this is the IPAddress they need to set as nexthop in UDR.

Reply:
Spoke with Yair, we will use the following description: The Firewall Internal Load Balancer IP to be used as the next hop in User Defined Routes